src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs

```diff
@@ -16,16 +17,29 @@ public partial class BinaryFormatterTests
     {
         private static void CheckForAnyEquals(object obj, object deserializedObj)
         {
-            if (obj != null && deserializedObj != null)
+            try
+            {
+                Assert.True(CheckEquals(obj, deserializedObj));
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Error during equality check of type " + obj?.GetType()?.FullName);
```

Comment:
Not necessary to change now, but it's better to use Assertion rather than doing Console.Writeline as the Console output is not associated with the test (and is not in the test log)

In this case it would basically be Assertion.IsTrue(false, ... string here...)

In the case where you want to add a message to a thrown AssertionException, you would throw a new AssertionException including the callstack of the old one.